dotCMS/src/main/java/com/dotcms/webdav/FsMemoryLockManager.java

```diff
@@ -0,0 +1,141 @@
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package com.dotcms.webdav;
+
+import com.dotcms.repackage.com.bradmcevoy.http.*;
+import com.dotcms.repackage.com.bradmcevoy.http.exceptions.NotAuthorizedException;
+import com.dotcms.repackage.com.ettrema.http.fs.LockManager;
+import com.dotmarketing.util.Logger;
+
+import java.io.File;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+
+/**
+ *
+ */
+public class FsMemoryLockManager implements LockManager {
+
+
+    /**
+     * maps current locks by the file associated with the resource
+     */
+    Map<File, CurrentLock> locksByFile;
+    Map<String, CurrentLock> locksByToken;
+
+    public FsMemoryLockManager() {
+        locksByFile = new HashMap<File, CurrentLock>();
+        locksByToken = new HashMap<String, CurrentLock>();
+    }
+
+    @Override
+    public synchronized LockResult lock( LockTimeout timeout, LockInfo lockInfo, LockableResource r ) {
+        FsResource resource = (FsResource) r;
+        LockToken currentLock = currentLock( resource );
+        if( currentLock != null ) {
+            return LockResult.failed( LockResult.FailureReason.ALREADY_LOCKED );
+        }
+
+        LockToken newToken = new LockToken( UUID.randomUUID().toString(), lockInfo, timeout );
+        CurrentLock newLock = new CurrentLock( resource.getFile(), newToken, lockInfo.lockedByUser );
+        locksByFile.put( resource.getFile(), newLock );
+        locksByToken.put( newToken.tokenId, newLock );
+        return LockResult.success( newToken );
+    }
+
+    @Override
+    public synchronized LockResult refresh( String tokenId, LockableResource resource ) {
+        CurrentLock curLock = locksByToken.get( tokenId );
+        if( curLock == null ) {
+          Logger.info(FileSystemResourceFactory.class, "can't refresh because no lock");
+            return LockResult.failed( LockResult.FailureReason.PRECONDITION_FAILED );
+        } else {
+            curLock.token.setFrom( new Date() );
+            return LockResult.success( curLock.token );
+        }
+    }
+
+    @Override
+    public synchronized void unlock( String tokenId, LockableResource r ) throws NotAuthorizedException {
+        FsResource resource = (FsResource) r;
+        LockToken lockToken = currentLock( resource );
+        if( lockToken == null ) {
+          Logger.info(FileSystemResourceFactory.class, "not locked" );
+            return;
+        }
+        if( lockToken.tokenId.equals( tokenId ) ) {
+            removeLock( lockToken );
+        } else {
+            throw new NotAuthorizedException( resource );
+        }
+    }
+
+    private LockToken currentLock( FsResource resource ) {
+        CurrentLock curLock = locksByFile.get( resource.getFile() );
+        if( curLock == null ) return null;

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Missing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00121)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00122)